src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx

```diff
@@ -1,201 +1,234 @@
-import React, { ChangeEvent, Fragment, useEffect, useState } from 'react'
+import React, { ChangeEvent, useEffect, useMemo } from 'react'
 import Checkbox from 'react-three-state-checkbox'
-import { Form, FormikProps } from 'formik'
 import { observer } from 'mobx-react-lite'
 
 import { FuncStepTypesEnum } from '@core/enum/func-step-types-enum'
 import { t } from '@i18n'
+import datasetStore from '@store/dataset'
 import filterStore from '@store/filter'
+import {
+  ConditionJoinMode,
+  TFuncCondition,
+  TVariant,
+} from '@service-providers/common/common.interface'
+import { IInheritanceModeCachedValues } from '../function-panel.interface'
+import functionPanelStore from '../function-panel.store'
 import { PanelButtons } from './panelButtons'
 
-export interface IInheritanceFormValues {
-  problemGroups: string[]
-  variants: string[]
-}
-
-export const InheritanceMode = observer(
-  ({
-    values,
-    setFieldValue,
-    submitForm,
-    resetForm,
-  }: FormikProps<IInheritanceFormValues>) => {
-    const cachedValues =
-      filterStore.readFilterCondition<IInheritanceFormValues>(
+export const InheritanceMode = observer(() => {
+  const cachedValues =
+    filterStore.readFilterCondition<IInheritanceModeCachedValues>(
+      FuncStepTypesEnum.InheritanceMode,
+    )
+
+  const { filterName } = functionPanelStore
```

Comment:
`const { filterName, filterGroup, problemGroups, complexVariants } = functionPanelStore `

Reply:
that was already fixed in https://github.com/ForomePlatform/Anfisa-React-Client/pull/438